pkg/azure/securitygroups.go

```diff
@@ -0,0 +1,263 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package azure
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-03-01/network"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	"github.com/submariner-io/cloud-prepare/pkg/k8s"
+)
+
+const (
+	internalSecurityGroupSuffix = "-nsg"
+	externalSecurityGroupSuffix = "-submariner-external-sg"
+	internalSecurityRulePrefix  = "Submariner-Internal-"
+	externalSecurityRulePrefix  = "Submariner-External-"
+	allNetworkCIDR              = "0.0.0.0/0"
+	basePriorityInternal        = 2500
+	baseExternalInternal        = 3500
+)
+
+type CloudInfo struct {
+	SubscriptionID string
+	InfraID        string
+	Region         string
+	BaseGroupName  string
+	Authorizer     autorest.Authorizer
+	K8sClient      k8s.Interface
+}
```

Comment:
Since this file defines `CloudInfo` and its methods I think the file should be named _cloud_info.go_.